momepy/preprocessing.py

```diff
@@ -756,6 +757,57 @@ def _polygonize_ifnone(edges, polys):
     return polys
 
 
+def _create_shape_index(gdf):
+    """
+    Shape index based on area and minimum_bounding_circle.
+    """
+    # Extract underlying PyGEOS geometries pygeos understands
+    # (minimum_bounding_circle is not yet exposed in geopandas)
+    ga = gdf.geometry.array.data
+    # measure area
+    gdf["area"] = gdf.area
+    # generate minimum bounding circle
+    circles = gpd.GeoDataFrame(
+        geometry=gpd.GeoSeries(pygeos.minimum_bounding_circle(ga), crs=gdf.crs)
+    )
+    # measure MBC's area
+    circles["circle_area"] = circles.area
+    # measure Reock (circular) compactness
+    gdf["reock"] = gdf["area"] / circles["circle_area"]
+    # measure direct shape index that captures banana-like relationship between area
+    # and Reock compactness in a one dimension
+    gdf["shape_index"] = gdf["area"] / np.sqrt(circles["circle_area"])
+
+    return gdf
+
+
+def _count_edges_per_polygon(lines_gdf, polygons_gdf):
+    """
+    Counts the number of linestrings that formed the polygons
+    """
+    # Check if the polygons are valid geometries and fix them if not
+    polygons_gdf["geometry"] = polygons_gdf["geometry"].apply(
+        lambda x: make_valid(x) if not x.is_valid else x
+    )
```

Comment:
This can be done as a vectorised call, `make_valid` is now available as `GeoSeries.make_valid`.